hibernate-entitymanager/src/main/java/org/hibernate/jpa/HibernateEntityManagerFactory.java

```diff
@@ -43,12 +46,40 @@
 	public SessionFactory getSessionFactory();
 
 	/**
-	 * Retrieve the EntityTypeImpl by name.  Use of the Hibernate O/RM notion the "entity name" allows support
+	 * Retrieve the {@code EntityTypeImpl} by name.  Use of the Hibernate O/RM notion the "entity name" allows support
 	 * for non-strictly-JPA models to be used in JPA APIs
 	 *
 	 * @param entityName The entity name
 	 *
-	 * @return The EntityTypeImpl
+	 * @return the {@code EntityTypeImpl}
 	 */
 	public EntityTypeImpl getEntityTypeByName(String entityName);
+
+	/**
+	 * Retrieve the {@code EntityGraphImpl} by name.
+	 *
+	 * @param name The entity name
+	 *
+	 * @return the {@code EntityGraphImpl}
+	 */
+	public EntityGraphImpl findEntityGraphByName(String name);
```

Comment:
Should that not really be `EntityGraph` instead of `EntityGraphImpl` which exposes internal classes. Same btw for `getEntityTypeByName`.